controllers/runtime.go

```diff
@@ -86,44 +84,64 @@ func logRuntimeInformation() {
 	log.Info("Runtime Information", "Proxy", RunInfo.Proxy)
 }
 
-func getRuntimeInformation(r *SpecialResourceReconciler) {
-
+func getRuntimeInformation(r *SpecialResourceReconciler) error {
 	var err error
 
-	err = cache.Nodes(r.specialresource.Spec.NodeSelector, false)
-	exit.OnError(errors.Wrap(err, "Failed to cache nodes"))
+	if err = cache.Nodes(r.specialresource.Spec.NodeSelector, false); err != nil {
+		return fmt.Errorf("failed to cache nodes: %w", err)
+	}
 
 	RunInfo.OperatingSystemMajor, RunInfo.OperatingSystemMajorMinor, RunInfo.OperatingSystemDecimal, err = cluster.OperatingSystem()
-	exit.OnError(errors.Wrap(err, "Failed to get operating system"))
+	if err != nil {
+		return fmt.Errorf("failed to get operating system: %w", err)
+	}
 
 	RunInfo.KernelFullVersion, err = kernel.FullVersion()
-	exit.OnError(errors.Wrap(err, "Failed to get kernel version"))
+	if err != nil {
+		return fmt.Errorf("failed to get kernel version: %w", err)
+	}
 
 	RunInfo.KernelPatchVersion, err = kernel.PatchVersion(RunInfo.KernelFullVersion)
-	exit.OnError(errors.Wrap(err, "Failed to get kernel patch version"))
+	if err != nil {
+		return fmt.Errorf("failed to get kernel patch version: %w", err)
+	}
 
 	// Only want to initialize the platform once.
 	if RunInfo.Platform == "" {
 		RunInfo.Platform, err = clients.Interface.GetPlatform()
-		exit.OnError(errors.Wrap(err, "Failed to determine platform"))
+		if err != nil {
+			return fmt.Errorf("failed to determine platform: %v", err)
+		}
 	}
 
 	RunInfo.ClusterVersion, RunInfo.ClusterVersionMajorMinor, err = cluster.Version()
-	exit.OnError(errors.Wrap(err, "Failed to get cluster version"))
+	if err != nil {
+		return fmt.Errorf("failed to get cluster version: %w", err)
+	}
 
 	RunInfo.ClusterUpgradeInfo, err = upgrade.ClusterInfo()
-	exit.OnError(errors.Wrap(err, "Failed to get upgrade info"))
+	if err != nil {
+		return fmt.Errorf("failed to get upgrade info: %w", err)
+	}
 
 	RunInfo.PushSecretName, err = retryGetPushSecretName(r)
-	warn.OnError(errors.Wrap(err, "Failed to get push secret name"))
+	if err != nil {
```

Comment:
This is a behavior change, but its better now, right? If there is an error retrieving the push secret name the helm chart might not be fully complete, so its best to bail out instead of rendering a bad template.